Mage/src/main/java/mage/abilities/effects/common/continuous/CantGainLifeRestOfGameTargetEffect.java

```diff
@@ -23,6 +27,26 @@ private CantGainLifeRestOfGameTargetEffect(final CantGainLifeRestOfGameTargetEff
         super(effect);
     }
 
+    @Override
+    public boolean applyToObjects(Layer layer, SubLayer sublayer, Ability source, Game game, List<MageItem> objects) {
```

Comment:
Must be `void applyToObjects` -- you must already collect fully and strictly affected objects before apply effect to it. There are impossible use cases with some objects list but not applied effect.

Also I recommend to replace `List<MageItem> objects` by `Map<uuid, MageItem>` to have fast access in inherited methods and logic.